File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-7.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int maxLen = 0;
        int left = 0;
        int n = s.length();
        int maxCount = 0;
        int count[26] = {0};
        int len = 0;

        for(int right=0; right < n; ++right)
        {
            count[s[right] - 'A']++;

            if(count[s[right] - 'A'] > maxCount)
            {
                maxCount = count[s[right] - 'A'];
            }

            while((right - left + 1) - maxCount > k)
            {
                count[s[left] - 'A']--;
                left++; 
            }

            len = right - left + 1;
            if(len > maxLen) maxLen = len;

        }

        return maxLen;
    }
};

File: Data Structures & Algorithms/same-binary-tree/submission-7.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        self.isSame: bool = True
        
        if not p and not q:
            return True
        
        def DFS(node1,node2):
            if node1 and node2:
                if node1.val != node2.val:
                    self.isSame = False
                    return
            elif (node1 and not node2) or (not node1 and node2):
                self.isSame = False
                return
            
            if node1.left and node2.left:
                DFS(node1.left,node2.left)
            elif (node1.left and not node2.left) or (not node1.left and node2.left):
                self.isSame = False
                return
            
            if node1.right and node2.right:
                DFS(node1.right,node2.right)
            elif (node1.right and not node2.right) or (not node1.right and node2.right):
                self.isSame = False
                return

        DFS(p,q)
        
        return self.isSame



            
